src/commonControls/numberControl/NumberControl.ts

```diff
@@ -550,850 +628,662 @@ export class NumberControl extends Control implements InteractionModelContributo
+    private handleChangeWithoutValue(input: ControlInput, resultBuilder: ControlResultBuilder): void {
+        this.askElicitationQuestion(input, resultBuilder, $.Action.Change);
+        return;
     }
 
-    private isFeedbackNoAndValueChangedWhenConfirmingValue(input: ControlInput): boolean {
+    private isBareValue(input: ControlInput): any {
         try {
             okIf(InputUtil.isValueControlIntent(input, AmazonBuiltInSlotType.NUMBER));
             const { feedback, action, target, values } = unpackValueControlIntent(
                 (input.request as IntentRequest).intent,
             );
             const valueStr = values[0].slotValue;
-            okIf(
-                InputUtil.actionIsMatchOrUndefined(action, [
-                    ...this.props.interactionModel.actions.set,
-                    ...this.props.interactionModel.actions.change,
-                ]),
-            );
+            okIf(InputUtil.feedbackIsUndefined(feedback));
+            okIf(InputUtil.actionIsUndefined(action));
             okIf(InputUtil.targetIsMatchOrUndefined(target, this.props.interactionModel.targets));
-            okIf(InputUtil.feedbackIsFalse(feedback));
             okIf(InputUtil.valueStrDefined(valueStr));
-            okIf(this.state.lastInitiative.actName === ConfirmValueAct.name);
-            okIf(this.state.value !== Number.parseInt(valueStr, 10));
-            this.handleFunc = this.handleFeedbackNoAndValueChangedWhenConfirmingValue;
+            this.handleFunc = this.handleBareValue;
             return true;
         } catch (e) {
             return falseIfGuardFailed(e);
         }
     }
 
-    private async handleFeedbackNoAndValueChangedWhenConfirmingValue(
-        input: ControlInput,
-        resultBuilder: ControlResultBuilder,
-    ): Promise<void> {
-        const { action, values } = unpackValueControlIntent((input.request as IntentRequest).intent);
-        const valueStr = values[0].slotValue;
+    private async handleBareValue(input: ControlInput, resultBuilder: ControlResultBuilder): Promise<void> {
+        const { valueStr } = InputUtil.getValueResolution(input);
         this.setValue(valueStr);
-        await this.commonHandlerWhenValueChanged(action ?? $.Action.Set, input, resultBuilder);
+        if (this.isConfirmationRequired(input) === true) {
+            this.addInitiativeAct(
+                new ConfirmValueAct(this, {
+                    value: this.state.value,
+                    renderedValue: this.props.valueRenderer(this.state.value, input),
+                }),
+                resultBuilder,
+            );
+        } else {
+            await this.validateAndAddActs(input, resultBuilder, this.state.elicitationAction ?? $.Action.Set);
+        }
+        return;
     }
 
-    private isFeedbackYesAndValueChangedWhenConfirmingValue(input: ControlInput): boolean {
+    private isConfirmationAffirmed(input: ControlInput): any {
```

Comment:
This is doing two things.  It would be best for the name to reflect this, or to break into two.

isConfirmationAffirmedOrAcceptanceOfSuggestion().

my inclination is to refactor into two handlers that each do one things.. it is code duplication but its easier to comprehend and maintain I think.

Reply:
Split into two bult-in handlers 
`isConfirmationAffirmed` and `isSuggestionAccepted`